src/canisters/frontend/ic-certified-assets/src/tests.rs

```diff
@@ -232,6 +240,59 @@ fn can_create_assets_using_batch_api() {
     );
 }
 
+#[test]
+fn serve_correct_encoding() {
+    let mut state = State::default();
+    let time_now = 100_000_000_000;
+
+    const IDENTITY_BODY: &[u8] = b"<!DOCTYPE html><html></html>";
+    const GZIP_BODY: &[u8] = b"this is 'gzipped' content";
+
+    create_assets(
+        &mut state,
+        time_now,
+        vec![
+            AssetBuilder::new("/contents.html", "text/html")
+                .with_encoding("identity", vec![IDENTITY_BODY]),
+            AssetBuilder::new("/contents.html", "text/html").with_encoding("gzip", vec![GZIP_BODY]),
+        ],
+    );
+
+    let identity_response = state.http_request(
+        RequestBuilder::get("/contents.html")
+            .with_header("Accept-Encoding", "identity")
+            .with_certificate_version(2)
+            .build(),
+        &[],
+        unused_callback(),
+    );
+    assert_eq!(identity_response.status_code, 200);
+    assert_eq!(identity_response.body.as_ref(), IDENTITY_BODY);
+
+    let gzip_response = state.http_request(
+        RequestBuilder::get("/contents.html")
+            .with_header("Accept-Encoding", "gzip")
+            .with_certificate_version(2)
+            .build(),
+        &[],
+        unused_callback(),
+    );
+    assert_eq!(gzip_response.status_code, 200);
+    assert_eq!(gzip_response.body.as_ref(), GZIP_BODY);
+
+    // V1 serves only the most important encoding - in this case "identity"
```

Comment:
AFAIK this is a change in behavior.  Given `Accept-Encoding: gzip`, and an asset with both identity and gzip encodings, the current asset canister returns the gzip encoding, but with the certification for the identity encoding 🤦‍♂️.  Is this change in behavior expected?

Reply:
Expected given the code changes, yes. I think it's ok to make a breaking change because this only affects users that 1) only accept `gzip` and nothing else, and 2) do not check for the certificate (over raw), and 3) use the HTTP interface, no other method, and 4) can't handle `identity` instead.

I can also offer these alternatives:
- Do not fix this bug in this PR and talk about that some more in e.g. an architecture meeting
- Revert to old behaviour if request is through `raw.ic0.app`

Reply:
Let's start with a description of the change in behavior in the changelog, with rationale.

Reply:
I modified the behaviour a bit, and documented it properly in the changelog. This should not break any clients, unless they check for the presence of a cert but don't validate the contents properly